trino/sqlalchemy/dialect.py

```diff
@@ -317,7 +317,7 @@ def _get_server_version_info(self, connection: Connection) -> Any:
         query = "SELECT version()"
         try:
             res = connection.execute(sql.text(query))
-            version = res.scalar()
+            version = res.scalar_one()
```

Comment:
Could you add a test?

Reply:
It seems that `351` behaves differently with query state ..

Reply:
cc: @arhimondr @losipiuk Is there consensus on whether the new behaviour is working as expected? I think this is related to https://github.com/trinodb/trino/pull/13055

Reply:
This should be a non-op change as the only difference between `scalar` and `scalar_one` is that `scalar_one` would additionally throw if no rows are returned.

Reply:
[SELECT version()](https://trino.io/docs/current/functions/system.html#version) should always return exactly one row and we want to raise exception when it doesn't.
Alternative solution might be to  execute `res.fetchall()` after `version = res.scalar()` but it's cumbersome.

Reply:
The issue is not limited to sqlalchemy. Following dbapi code will also result in cancelled query while the result is scalar.

```
cur.execute('SELECT VERSION()')
cur.fetchone()
cur.cancel()
```

It is indeed related to the PR on Trino that @hashhar refers to. The issue is that the python client doesn't acknowledge the reception of the data by calling the next_uri on the Trino API.

Reply:
I believe https://github.com/trinodb/trino-python-client/pull/220 is the way to go. Thanks @mdesmet for the thorough investigation and the fix.